File: app/src/main/java/ga/chrom_web/player/multiplayer/SharedPreferenceHelper.java
package ga.chrom_web.player.multiplayer;


import android.content.SharedPreferences;
import android.graphics.Color;

import javax.inject.Inject;

public class SharedPreferenceHelper {

    private static final String PREF_NICK = "nick";
    private static final String PREF_COLOR = "color";

    @Inject
    SharedPreferences prefs;

    public SharedPreferenceHelper() {
         App.getComponent().inject(this);
    }

    public void saveUser(String nick, int color) {
        // TODO: maybe save color in String ???
        SharedPreferences.Editor editor = prefs.edit();
        editor.putString(PREF_NICK, nick);

        editor.putInt(PREF_COLOR, removeAlpha(color));
        editor.apply();
    }

    private int removeAlpha(int color) {
        String hexColor = Integer.toHexString(color);
        if (hexColor.length() <= 6) {
            return color;
        }
        String colorWithoutAlpha = "#" + hexColor.substring(2);
        return Color.parseColor(colorWithoutAlpha);
    }

    public String getNick() {
        return prefs.getString(PREF_NICK, "DEFAULT");
    }

    public int getColor() {
        return prefs.getInt(PREF_COLOR, 0);
    }

    public String getHexColor() {
        return "#" + Integer.toHexString(getColor());
    }
}

File: app/src/main/java/ga/chrom_web/player/multiplayer/SharedPreferenceHelper.kt
package ga.chrom_web.player.multiplayer


import android.content.SharedPreferences
import android.graphics.Color

import javax.inject.Inject

import ga.chrom_web.player.multiplayer.di.App

class SharedPreferenceHelper {

    companion object {
        private val PREF_NICK = "nick"
        private val PREF_COLOR = "color"
        private val PREF_SERVER_SMILES_PATH = "server_smiles_path"
        private val PREF_SMILES_VERSION = "smiles_version"
    }

    @Inject
    lateinit var prefs: SharedPreferences

    val nick: String
        get() = prefs.getString(PREF_NICK, "DEFAULT")

    val color: String
        get() = prefs.getString(PREF_COLOR, "#000000")

    val smilesServerPath: String
        get() = prefs.getString(PREF_SERVER_SMILES_PATH, "")

    val smilesVersion: Int
        get() = prefs.getInt(PREF_SMILES_VERSION, 1)

    init {
        App.getComponent().inject(this)
    }

    fun saveUser(nick: String, color: Int) {
        // TODO: maybe save color in String ???
        val editor = prefs.edit()
        editor.putString(PREF_NICK, nick)
        editor.putString(PREF_COLOR, removeAlpha(color))
        editor.apply()
    }

    fun saveSmilesServerPath(path: String) {
        prefs.edit().putString(PREF_SERVER_SMILES_PATH, path).apply()
    }

    /**
     * Checks if server has new version smiles
     * @return true if version of smiles updated,
     *         false when version same
     */
    fun isNewerSmilesVersion(version: Int): Boolean {
        if (smilesVersion != version) {
            prefs.edit().putInt(PREF_SMILES_VERSION, version).apply()
            return true
        }
        return false
    }

    private fun removeAlpha(color: Int): String {
        val hexColor = Integer.toHexString(color)
        if (hexColor.length <= 6) {
            return "#" + hexColor;
        }
        val colorWithoutAlpha = "#" + hexColor.substring(2)
        return colorWithoutAlpha
    }
}
